### PR TITLE
Add docs for IDE0110.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Corresponding docs PR for https://github.com/dotnet/roslyn/pull/49141.

### CODE REVIEW

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:27]: Can the hyperlink point to a documentation page for discards instead of the specification? Tag @gewarren

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:19]:
```suggestion
| **Rule ID** | IDE0110 |
```

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:10]: It is fine to put your alias if you plan to maintain/fix this doc page. Otherwise, I would recommend changing the author and ms.author to `gewarren`, so the docs team owns it going forward.

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:11]:
```suggestion
author: gewarren
ms.author: gewarren
```

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:2]: I think we should keep the title consistent with other unnecessary code rules, i.e. `Remove unnecessary discard`
```suggestion
title: "IDE0110: Remove unnecessary discard"
```

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:3]:
```suggestion
description: "Learn about code analysis rule IDE0110: Remove unnecessary discard"
```

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:15]:
```suggestion
# Remove unnecessary discard (IDE0110)
```

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:20]:
```suggestion
| **Title** | Remove unnecessary discard |
```

File: docs/fundamentals/code-analysis/style-rules/index.md
Comment (mavasani) [inline docs/fundamentals/code-analysis/style-rules/index.md:113]:
```suggestion
> | [IDE0110](ide0110.md) | Remove unnecessary discard | |
```

**Review by mavasani — APPROVED** [review]

**Comment by mavasani** [comment]
@CyrusNajmabadi thanks a lot for the PR! I would suggest waiting for @gewarren's review before merging.

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (BillWagner) [reply docs/fundamentals/code-analysis/style-rules/ide0110.md:2]: Yes, we have evidence that users search on the code rule text.

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (BillWagner) [reply docs/fundamentals/code-analysis/style-rules/ide0110.md:10]: Related tip: If you leave them blank, the default owner (in this case @gewarren) will be used.

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (BillWagner) [reply docs/fundamentals/code-analysis/style-rules/ide0110.md:27]: Two comments here. First, I've added a note to #20584 that the discard pattern should be covered. (It wasn't called out in the issue, and our current language reference doesn't cover it explicitly).

Second, for now, let's update the link to point to the pattern spec that's published on docs. This helps me in two ways. First, as we get to the updated language reference, I'll find the link and replace it. Second, when we intend to link to the proposals, I'll find and update those links when the proposal text moves to the official standard. (Bonus nit: this suggestion removes the whitespace that's causing the markdown lint warning.

```suggestion
This rule flags unnecessary [discard patterns](../../_csharplang/proposals/csharp-8.0/patterns.md#discard-pattern), when this discard pattern is used in a context where it has no effect.
```

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (CyrusNajmabadi) [reply docs/fundamentals/code-analysis/style-rules/ide0110.md]: sgtm.

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi thanks a lot for the PR! I would suggest waiting for @gewarren's review before merging.

Will do, thanks.

File: docs/fundamentals/code-analysis/style-rules/ide0110.md
Comment (BillWagner) [inline docs/fundamentals/code-analysis/style-rules/ide0110.md:27]: my bad. I had the wrong number of folders:

```suggestion
This rule flags unnecessary [discard patterns](../../../../_csharplang/proposals/csharp-8.0/patterns.md#discard-pattern), when this discard pattern is used in a context where it has no effect.
```

**Review by gewarren — APPROVED** [review]

File: docs/fundamentals/code-analysis/style-rules/unnecessary-code-rules.md
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/style-rules/unnecessary-code-rules.md:37]: Following the same pattern for IDE0080 and IDE0081

```suggestion
- [Remove unnecessary discard (IDE0110)](ide0110.md) - C# only.
```

**Pull request merged**
